apollo-router/src/services/layers/persisted_queries/mod.rs

```diff
@@ -175,33 +171,20 @@ impl PersistedQueryLayer {
                 return Ok(request);
             }
 
-            match context_guard.get::<Compiler>() {
+            match context_guard.get::<ParsedDocument>() {
                 None => {
                     drop(context_guard);
-                    // For some reason, QueryAnalysisLayer didn't give us a Compiler?
+                    // For some reason, QueryAnalysisLayer didn't give us a document?
                     return Err(supergraph_err(
                         graphql_err(
                             "MISSING_PARSED_OPERATION",
-                            "internal error: compiler missing",
+                            "internal error: executable document missing",
                         ),
                         request,
                         ErrorCacheStrategy::DontCache,
                     ));
                 }
-                Some(c) => c.0.clone(),
-            }
-        };
-
-        let compiler_guard = compiler.lock().await;
-        let db = &compiler_guard.db;
-        let file_id = match db.source_file(QUERY_EXECUTABLE.into()) {
```

Comment:
Is the operation now always present? (lot of context i m missing so just asking ^^)

Reply:
Before this PR:

* Router context private entries contain `Arc<Mutex<ApolloCompiler>>`
* `ApolloCompiler` contains any number of files, identified by `FileId`, each with an associated "file name"
* This particular line looks for a file named `"query.graphql"` and finds its ID. We expect it to always exist so `unwrap` would not be wrong, but in case of a bug causing it to be missing we return an error instead of panicking.
* With that ID, we can later query the equivalent of `Arc<ExecutableDocument>` from the compiler

After this PR:

* Router context private entries contain `Arc<ParsedDocument>`
* `ParsedDocument` (a new Router struct) contains `ExecuableDocument`

File IDs still exist in apollo-compiler 1.0 but users almost never need to care about them.